### PR TITLE
variable substitution added to policy's mode: role: 

### PR DESCRIPTION
**Pull request by FireballDWF**
{{account_id}} in a lambda based policy mode: role: will expand to the AWS account id where the Lambda function is being deployed

Example:
      role: "arn:aws:iam::{{account_id}}:role/lambda-cloudcustodian"

Benefit: can now write a policy once then deploy to multiple accounts

### CODE REVIEW

File: c7n/policy.py [hunk 1]
```diff
@@ -470,6 +471,15 @@ def run(self, event, lambda_context):
                 self.policy._write_file(
                     "action-%s" % action.name, utils.dumps(results))
         return resources
+    def expand_variables(self, policy):
+        """expand any variables in the action to_from/cc_from fields.
```
Comment (kapilt) [inline c7n/policy.py]: doc string is referencing the notify origin, worth updating to its purpose here.

File: c7n/policy.py [hunk 2]
```diff
@@ -470,6 +471,15 @@ def run(self, event, lambda_context):
...
+        """
+        p = self.policy.data.copy()
+        if 'mode' in self.policy.data:
+            mode = self.policy.data['mode'].copy()
+            mode['role'] = re.sub(r'{{account_id}}',policy['account_id'],mode['role'])
```
Comment (kapilt) [inline c7n/policy.py]: any reason for the sub vs string format here? also need to check conditionally check if role is specified, we typically use --assume on the cli to pick this up for cross account usage.

File: c7n/policy.py [hunk 3]
```diff
@@ -478,6 +488,14 @@ def provision(self):
         with self.policy.ctx:
             self.policy.log.info(
                 "Provisioning policy lambda %s", self.policy.name)
+
+            policy = {
```
Comment (kapilt) [inline c7n/policy.py]: naming this as policy is perhaps a little confusing, config/vars/etc would be more apt.

**Comment by dan-v** [comment]
I'd love to have this feature. Anything I can do to help get this change pushed through?

**Comment by FireballDWF** [comment]
new commit fixes the link/travis issue : https://github.com/FireballDWF/cloud-custodian/commit/cef685ab5c23c7f41edc1fe364fb8975dc6315a5

**Comment by kapilt** [comment]
merged in #1559

**Pull request closed without merging**